jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/MimeTypes.java

```diff
@@ -797,6 +846,84 @@ else if (!Objects.equals(value, otherValue))
         }
     }
 
+    /**
+     * @deprecated Create a new {@link MimeTypes} with your settings, and call {@link #setFrom(MimeTypes)} instead.
+     */
+    @Deprecated(since = "12.1.0", forRemoval = true)
+    public static class Wrapper extends Mutable
+    {
+        private MimeTypes _wrapped;
+
+        public Wrapper()
+        {
+            super(null);
+        }
+
+        public MimeTypes getWrapped()
+        {
+            return _wrapped;
+        }
+
+        public void setWrapped(MimeTypes wrapped)
+        {
+            _wrapped = wrapped;
+        }
+
+        @Override
+        public String getMimeForExtension(String extension)
+        {
+            String mime = super.getMimeForExtension(extension);
+            return mime == null && _wrapped != null ? _wrapped.getMimeForExtension(extension) : mime;
```

Comment:
This logic seems really wrong.

If there is a non-null `_wrapped`, but the call to `super` returns non-null, then the `_wrapped` is ignored.

It's a wrapper, so the logic should rather be;

```java
if (_wrapped == null)
    return super.getMimeForExtension(extension);
else
    return _wrapped.getMimeForExtension(extension);
```

And same for the other methods.

Reply:
This code is literally what was there before the class was removed.
Wasn't keen on changing behavior.

Reply:
Actually, your proposal changes the behavior of inferred charsets in the test cases.

``` java
MimeTypes.Mutable wrapped = new MimeTypes.Mutable(null);
wrapper.setWrapped(wrapped);
assertThat(wrapper.getCharsetInferredFromContentType("text/plain"), equalToIgnoringCase("us-ascii")); // THIS FAILS
```